### ISSUE TITLE
Data error for UK

### ISSUE THREAD

**Issue by DrHyde** [googlei18n/libphonenumber#1390]
You provide as an exampleNumber for tollFree numbers in the UK '8001234567'. This number is invalid. The 800123 range is not in use, and when it was in use it was for 9 digit numbers. You could use 8001223456 instead.

[Source](http://static.ofcom.org.uk/static/numbering/S8.xls) (excel file of number allocations and lengths)

**Comment by DrHyde** [comment]
Similarly 5612345678 that you give for VOIP numbers is not valid. 5600001234 is OK as a replacement.

**Comment by DrHyde** [comment]
And 5512345678. Try 5511001234 instead.

[Source](http://static.ofcom.org.uk/static/numbering/S5.xls)

**Comment by DrHyde** [comment]
And 1481456789 for GG (Guernsey) which isn't allocated. Try 1481201234.

[Source](http://static.ofcom.org.uk/static/numbering/sabcde14.xls)

**Comment by giggsey** [comment]
I'm assuming that this also means the metadata is incorrect too?

**Comment by DrHyde** [comment]
Quite possibly :-(

**Comment by DrHyde** [comment]
And 1624456789 for IM (Isle of Man), try 1624500123.

And 1534456789 for JE (Jersey), try 1534281234.

And 7797123456 for JE, try 7797712345.

Some of this might come down to a difference in interpretation of what is valid. At least some of these *could* exist but they've just not been allocated (that's the case for all of the ones for GG/IM/JE), so the metadata could be considered correct but the example just can't be dialled.

**Comment by kkeshava** [comment]
Hi,
Thank you for taking the time to improve libphonenumber!

We are aware of these updates that needs to be incorporated to metadata of UK. We are already working on it, as it’s a huge update it might take some time for us to fix.

**Comment by stepio** [comment]
We're also interested in this fix.
This phone number is detected as invalid, although it's not: +447700954321

**Comment by stepio** [comment]
Hmmm... Sorry, seems I was wrong.
https://fakenumber.org/phone/+44-7700-900946

But it was marked as valid by another online validator.

**Comment by penmetsaa** [comment]
@stepio,
Thank you for taking the time to improve libphonenumber!

We can't connect to +44 77009 54321. Do you have any supporting evidence that says 77009 numbers are valid? If so please raise separate GitHub issue. This issue is about updating library based on the complete numbering plan of +44 regions (GB, GG, IM, JE).

Please check our [contribution guidelines](https://github.com/googlei18n/libphonenumber/blob/master/CONTRIBUTING.md).

**Comment by DrHyde** [comment]
+44 7700 900 NNN are reserved for use in drama, and so valid but not reachable. The rest of +44 7700 9 may be allocated at some point in the future which is why some validators may say that it's *valid*, if they distinguish between valid and allocated.

**Comment by penmetsaa** [comment]
@DrHyde ,
Thanks for the update. :) So we are not adding support to any of 7700 9 rages as library considers only assigned ranges as valid ones. 

Progress done so far to support all docs in  http://www.ofcom.org.uk/static/numbering/:
1. Support to Mobile & Pager numbers in http://www.ofcom.org.uk/static/numbering/S7.xls: Done as part of release 8.1.0
2. Fixed-line ranges in http://static.ofcom.org.uk/static/numbering/#geog1: Will be part of next release.
3. All other docs and categories: In progress.

**Comment by penmetsaa** [comment]
Some updates on issues that reported here.
1. Already fixed.
2. Support to Fixed-line ranges: Done. Fixed as part of release 8.4.0
3. Support all other docs and categories: **In progress.**
4. Fine grained GG, IM and JE fixed-line ranges so that we support fixed-line ranges that are only live. This fixes issues mentioned in comments [#1390#issuecomment-254607027](https://github.com/googlei18n/libphonenumber/issues/1390#issuecomment-254607027), [#1390#issuecomment-286796519](https://github.com/googlei18n/libphonenumber/issues/1390#issuecomment-286796519) and [#1390#issuecomment-254611617](https://github.com/googlei18n/libphonenumber/issues/1390#issuecomment-254611617). This is part of release 8.4.3

**Comment by penmetsaa** [comment]
Hi DrHyde@,

Regarding your original request i.e 800123 range is not in use. As per below doc it is mentioned that is assigned already. http://static.ofcom.org.uk/static/numbering/S8.xls. And these numbers can be 9 or 10 digits in length as per [Wikpedia](https://en.wikipedia.org/wiki/Non-geographic_telephone_numbers_in_the_United_Kingdom#Freephone_-_including_0800_and_0808).

You are right in case of VOIP number (5612345678) and UAN number (5512345678) . 5612 and 5512 numbers are protected as per this doc. http://static.ofcom.org.uk/static/numbering/S5.xls

We will make these changes as part of this bug and update accordingly.

**Comment by DrHyde** [comment]
Yeah, I misunderstood what "closed range" means, so some numbers in that range may be in use. Nevertheless, OFCOM (not Wikipedia) is authoritative and says that it is 9 digits.

**Comment by penmetsaa** [comment]
@DrHyde,

Yes, you are right regarding the length. There are no valid numbers online that are 10 digits in length under range 800 123. We already have a task in pipeline to cleanup the lengths in our GB metadata based on OFCOM  doc http://static.ofcom.org.uk/static/numbering/S8.xls

**Comment by andyst** [comment]
I think the only / main remaining part is making the pattern / ranges for` <voip>` more granular, isn't it?

Updates since the last comment:

- Updated short codes in GB, GG, JE and IM.
- Remove 500 toll free numbers from United Kingdom and Guernsey (GB, GG, +44).
- Isle of Man (IM, +44): Move prefixes 7624[01689] from Pager to Mobile.
- Isle of Man (IM, +44): Move prefixes 7624[2-4] from Pager to Mobile.
- Support 10 digit 7624[0134689] pager numbers in IM region. And remove 7624[013-9] pager ranges from GB region.
